src/charm.py

```diff
@@ -471,6 +457,42 @@ def _on_update_status(self, _: UpdateStatusEvent) -> None:
         """Handle the update of charm status."""
         self._ensure_reconcile_timer_is_active()
         self._log_juju_processes()
+        self._log_charm_metrics()
+
+    @staticmethod
+    def _log_juju_processes() -> None:
+        """Log the running Juju processes.
+
+        Log all processes with 'juju' in the command line.
+        """
+        try:
+            processes, _ = execute_command(
+                ["ps", "afuwwx"],
+                check_exit=True,
+            )
+            juju_processes = "\n".join(line for line in processes.splitlines() if "juju" in line)
+            logger.info("Juju processes: %s", juju_processes)
+        except SubprocessError:
+            logger.exception("Failed to get Juju processes")
+
+    def _log_charm_metrics(self) -> None:
```

Comment:
Maybe better 
```suggestion
    def _log_charm_status(self) -> None:
```
to not confuse it with the other charm metric such as reconciliation (I know, in the future this will be a workload metric).

Reply:
not really, I just want the charm revision. Updated to the new name